Automattic/woocommerce-payments
# Adding Document list to WCPay client

Fixes #3929 
Fixes #3930

#### Changes proposed in this Pull Request

* Add feature flag for Documents list.
  * Updated dev tools to have checkbox to enable Documents: 53-gh-Automattic/woocommerce-payments-dev-tools
* Add components for merchants to be able to view their account Documents.
  * Add data store for documents.
  * Add filters for the document listing.
  * Add documents menu option under Payments. This should be temporary until tabbed page is built for settings.
* Add `WC_REST_Payments_Documents_Controller` to handle requests to serve the documents to the components.
* Update the API client to handle requests for documents.
* Update tests accordingly.

#### Testing instructions

* Add a few documents to your account with the CLI commands below, making sure to replace `<id>` with your account or blog ID.
  * Once added, go into the `wcpay_test_documents` table and edit one of the documents with the date of `2022-01-01` to be the `type` of `null_document`.
* Enable the documents list in the dev tools, which may require an update to the dev tools to get the checkbox.
* Navigate to Payments > Documents.
* Confirm the documents are listed with their Date, Type, and Description.
  * The `null_document` should have `Unknown document type` for both the Type and Description.
  * Documents with a `period_from/to` should have those respective dates listed in their description.
* Confirm sorting by date works.
* Test the filters under Show > Advanced filters:
  * Confirm _Date_ filters for _Before_, _After_, and _Between_ function as expected.
  * Confirm _Type_ filters with _Is_ and _Is not_ function as expected.
  * Confirm _All_ and _Any_ match settings function as expected.
* Navigate to WCPay Dev tools
* Update _Redirect API requests to:_ to an invalid url. (I just add `error` to the end of the url)
* Navigate back to Payments > Documents
* A snackbar error should appear in the lower left stating: _Error retrieving documents._


```
wp wcpay add_document_for_account <id> test_document --date=2022-01-01
wp wcpay add_document_for_account <id> test_document --date=2022-01-01
wp wcpay add_document_for_account <id> test_document --date=2022-02-01 --period_from="2022-01-01 00:00:00" --period_to="2022-01-31 23:59:59"
wp wcpay add_document_for_account <id> test_document --date=2022-03-01 --period_from="2022-02-01 00:00:00" --period_to="2022-02-28 23:59:59"
```

-------------------


**Post merge**
